adam/flask_backend/app.py

```diff
@@ -135,7 +143,116 @@ def get_scene() -> Any:
     }
 
 
+@app.route("/api/get_image", methods=["GET"])
+def get_image() -> Any:
+    if not request.args:
+        abort(HTTPStatus.BAD_REQUEST)
+
+    image_path = request.args.get("image_path", default="").strip("/")
+    image_file = DATA_DIR.joinpath(image_path).resolve()
+    if not is_relative_to(image_file, DATA_DIR):
+        abort(HTTPStatus.BAD_REQUEST)
+    print(image_file)
+    with open(image_file, "rb") as fp:
+        image_data = b64encode(fp.read()).decode("utf-8")
+    return {"image_data": image_data}
+
+
+@app.route("/api/poll_changes", methods=["GET"])
+def poll_changes() -> Any:
+    if not request.args:
+        abort(HTTPStatus.BAD_REQUEST)
+    learner = request.args.get("learner", default="")
+    training_curriculum = request.args.get("training_curriculum", default="")
+    testing_curriculum = request.args.get("testing_curriculum", default="")
+    if not learner and not training_curriculum and not testing_curriculum:
+        abort(HTTPStatus.BAD_REQUEST)
+
+    if any("/" in arg for arg in (learner, training_curriculum, testing_curriculum)):
+        abort(HTTPStatus.BAD_REQUEST)
+
+    curriculum_dir = (
+        LEARNERS_DIR
+        / learner
+        / EXPERIMENTS_DIR_NAME
+        / training_curriculum
+        / EXPERIMENTS_TESTING_DIR_NAME
+        / testing_curriculum
+    ).resolve()
+
+    if not is_relative_to(curriculum_dir, LEARNERS_DIR):
+        abort(HTTPStatus.BAD_REQUEST)
+
+    if (
+        not app.config["LEARNER_DIR_CONTENTS"].get(learner)
+        or not app.config["LEARNER_DIR_CONTENTS"][learner].get(training_curriculum)
+        or not app.config["LEARNER_DIR_CONTENTS"][learner][training_curriculum].get(
+            testing_curriculum
+        )
+        or not curriculum_dir.is_dir()
+    ):
+        return {"status": False, "message": "Invalid curriculum"}
+
+    ref_experiments = app.config["LEARNER_DIR_CONTENTS"][learner][training_curriculum][
+        testing_curriculum
+    ]
+    update_message = []
+    for situation_dir in curriculum_dir.iterdir():
+        if "situation" in situation_dir.name and situation_dir.is_dir():
+            if situation_dir.name not in ref_experiments:
+                update_message.append(
+                    f"New decode available for situation {int(situation_dir.name.rsplit('_')[-1]) + 1}"
+                )
+                ref_experiments[situation_dir.name] = set(
+                    file.name for file in situation_dir.iterdir()
+                )
+            else:
+                for file in situation_dir.iterdir():
+                    if file.name not in ref_experiments[situation_dir.name]:
+                        update_message.append(
+                            f"Updated information available for {int(situation_dir.name.rsplit('_')[-1]) + 1}: {file.name}"
```

Comment:
Missing a word here:
```suggestion
                            f"Updated information available for situation {int(situation_dir.name.rsplit('_')[-1]) + 1}: {file.name}"
```